extensions/EngineProcedureEXT.txt

```diff
@@ -0,0 +1,34 @@
+EngineProcedureEXT - Allow clients to wrap engine procedure
+
+About
+-----
+Clients may be interested in having more control over the engine procedure. In
+particular, they may want to modify or access the generated output before
+passing it to the underlying platform. Some clients, like Wine, may have
+thread requirements that SDL's threading API cannot satisfy. This extension
```

Comment:
In addition to SDL not giving us enough control over the thread, certain platforms that generate their own audio thread also conflict with situations like Wine and BRUTE. For formality's sake, we can replace "SDL's threading" with "the platform's audio library".

Everything else looks good to me, just need `EXT` suffixes on the new types and functions.